src/csharp/Grpc.IntegrationTesting/CustomErrorDetailsTest.cs

```diff
@@ -0,0 +1,112 @@
+    /// </summary>
+    public class CustomErrorDetailsTest
+    {
+        const string DebugInfoTrailerName = "debug-info-bin";
+        const string ExceptionDetail = "Exception thrown on purpose.";
+        const string Host = "localhost";
+        Server server;
+        Channel channel;
+        TestService.TestServiceClient client;
+
+        [TestFixtureSetUp]
+        public void Init()
+        {
+            // Disable SO_REUSEPORT to prevent https://github.com/grpc/grpc/issues/10755
+            server = new Server(new[] { new ChannelOption(ChannelOptions.SoReuseport, 0) })
+            {
+                Services = { TestService.BindService(new CustomErrorDetailsTestServiceImpl()) },
+                Ports = { { Host, ServerPort.PickUnused, ServerCredentials.Insecure } }
+            };
+            server.Start();
+
+            channel = new Channel(Host, server.Ports.Single().BoundPort, ChannelCredentials.Insecure);
+            client = new TestService.TestServiceClient(channel);
+        }
+
+        [TestFixtureTearDown]
+        public void Cleanup()
+        {
+            channel.ShutdownAsync().Wait();
+            server.ShutdownAsync().Wait();
+        }
+
+        [Test]
+        public async Task UnaryCall()
+        {
+            var call = client.UnaryCallAsync(new SimpleRequest { ResponseSize = 10 });
+
+            try
+            {
+                await call.ResponseAsync;
+                Assert.Fail();
+            }
+            catch (RpcException e)
+            {
+                Assert.AreEqual(StatusCode.Internal, e.Status.StatusCode);
+                var debugInfo = GetDebugInfo(call.GetTrailers());
+                Assert.AreEqual(debugInfo.Detail, ExceptionDetail);
+                Assert.IsNotEmpty(debugInfo.StackEntries);
+            }
+        }
+
+        private DebugInfo GetDebugInfo(Metadata trailers)
+        {
+            var entry = trailers.First((e) => e.Key == DebugInfoTrailerName);
+            return DebugInfo.Parser.ParseFrom(entry.ValueBytes);
+        }
+
+        private class CustomErrorDetailsTestServiceImpl : TestService.TestServiceBase
+        {
+            public override async Task<SimpleResponse> UnaryCall(SimpleRequest request, ServerCallContext context)
+            {
+                try
+                {
+                    throw new ArgumentException(ExceptionDetail);
+                }
+                catch (Exception e)
+                {
+                    // Fill debug info with some structured details about the failure.
+                    var debugInfo = new DebugInfo();
+                    debugInfo.Detail = e.Message;
+                    debugInfo.StackEntries.AddRange(e.StackTrace.Split(new[] { Environment.NewLine }, StringSplitOptions.None)); 
+                    context.ResponseTrailers.Add(DebugInfoTrailerName, debugInfo.ToByteArray());
+                    throw new RpcException(new Status(StatusCode.Internal, "The handler threw exception."));
```

Comment:
Is there a standard practice to return the debug-info trailer on INTERNAL status codes?

Optional nit:
If not, I might change this status code to something like INVALID_ARGUMENT. My thinking is just that INVALID_ARGUMENT might be more useful than INTERNAL here, and it might be helpful to make clear that it's possible to use it here.